apps/src/codegen.js

```diff
@@ -111,10 +111,44 @@ exports.workspaceCode = function(blockly) {
 };
 
 //
+// Property access wrapped in try/catch. This is in an indepedendent function
+// so the JIT compiler can optimize the calling function.
+//
+
+function safeReadProperty(object, property) {
```

Comment:
under what conditions is an exception thrown here? It seems like you might be able to avoid an exception by testing for the object or property's existence first.